src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,133 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import { Todo } from '../../types/Todo';
+import classNames from 'classnames';
+import { deleteTodo } from '../../api/todos';
+import callError from '../../utils/callError';
+import { MainContext } from '../../ContextProvider/ContextProvider';
+
+type TodoProps = {
+  todo: Todo;
+};
+
+const TodoItem: React.FC<TodoProps> = ({ todo }) => {
+  const context = useContext(MainContext);
+  const { todos, setTodos, setError, loadingIds } = context;
+
+  const { id, title, completed } = todo;
+
+  const [todoState, setTodoState] = useState({
+    isEdited: false,
+    editedValue: title,
+    completed: completed,
+  });
+
+  const isLoading = useRef(loadingIds.some(x => x === id));
+
+  if (isLoading.current) {
+    setTimeout(() => (isLoading.current = false), 3000);
+  }
+
+  useEffect(() => {
```

Comment:
you don't need this

Instead, you can add `onBlur` listener to input. And on blur just do some action (but this part in the next task) and change `isEdited` to false